drivers/pwm/pwm_led_esp32.c

```diff
@@ -361,15 +369,21 @@ static const struct pwm_driver_api pwm_led_esp32_api = {
 
 PINCTRL_DT_INST_DEFINE(0);
 
-#define CHANNEL_CONFIG(node_id)                                                \
-	{                                                                      \
-		.idx = DT_REG_ADDR(node_id),                                   \
-		.channel_num = DT_REG_ADDR(node_id) % 8,                       \
-		.timer_num = DT_PROP(node_id, timer),                          \
-		.speed_mode = DT_REG_ADDR(node_id) < SOC_LEDC_CHANNEL_NUM      \
-				      ? LEDC_LOW_SPEED_MODE                    \
-				      : !LEDC_LOW_SPEED_MODE,                  \
-		.clock_src = CLOCK_SOURCE,                                     \
+#if SOC_LEDC_SUPPORT_APB_CLOCK
+	#define CLOCK_SOURCE	LEDC_APB_CLK
+#elif SOC_LEDC_SUPPORT_PLL_DIV_CLOCK
+	#define CLOCK_SOURCE	LEDC_SCLK
+#endif
+
+#define CHANNEL_CONFIG(node_id)                                                                    \
+	{                                                                                          \
+		.idx = DT_REG_ADDR(node_id),                                                       \
+		.channel_num = DT_REG_ADDR(node_id) % 8,                                           \
+		.timer_num = DT_PROP(node_id, timer),                                              \
+		.speed_mode = DT_REG_ADDR(node_id) < SOC_LEDC_CHANNEL_NUM ? LEDC_LOW_SPEED_MODE    \
+									  : !LEDC_LOW_SPEED_MODE,  \
+		.clock_src = CLOCK_SOURCE,                                                         \
+		.inverted = DT_PROP(node_id, init_high),                                           \
```

Comment:
I couldn't be sure this flag would fit in the field `flags` already foreseen in the DT (pwm_dt_spec / pwm_flags_t)
It would align better with the runtime check made in function `set_cycles`

Reply:
Do you know how to get the value of the flags field here?
I‘d actually prefer that over adding an extra dt-property but I didn’t find a way to do it.

Reply:
in `include/zephyr/drivers/pwm.h` stay all the macros to unpack dts values:
```
 *    const struct pwm_dt_spec spec =
 *        PWM_DT_SPEC_GET_BY_NAME(DT_NODELABEL(n), alpha);
 *
 *    // Initializes 'spec' to:
 *    // {
 *    //         .dev = DEVICE_DT_GET(DT_NODELABEL(pwm1)),
 *    //         .channel = 1,
 *    //         .period = 1000,
 *    //         .flags = PWM_POLARITY_NORMAL,
 *    // }
 ```
 does it make sense to declare a pwms field for this driver (and include the flag)?
 I don't know enough about the API to give a solid opinion now, but others might know better.
 
 ```
 pwms = <&pwm2 3 2000 PWM_POLARITY_INVERTED>;
 ```